crates/uv-auth/src/store.rs

```diff
@@ -75,41 +20,64 @@ impl Default for AuthenticationStore {
 impl AuthenticationStore {
     pub fn new() -> Self {
         Self {
-            credentials: Mutex::new(HashMap::new()),
+            store: Mutex::new(HashMap::new()),
         }
     }
 
-    pub fn get(&self, url: &Url) -> Option<Option<Credential>> {
+    /// Retrieve stored credentials for a URL, if any.
+    pub fn get(&self, url: &Url) -> Option<Arc<Credentials>> {
```

Comment:
Adds borrowing of credentials instead of cloning.

Avoids the nested `Option<Option<...>>` use `should_attempt_fetch` instead.